libs/framework/src/celix_bundle_archive.c

```diff
@@ -288,206 +284,94 @@ celix_status_t celix_bundleArchive_create(celix_framework_t* fw, const char *arc
         } while (0);
     }
     if (status != CELIX_SUCCESS) {
-        error = "Failed to setup archive paths.";
-        goto init_failed;
+        fw_log(fw->logger, CELIX_LOG_LEVEL_ERROR, "Failed to create bundle archive; Out of memory.");
+        return status;
     }
 
-    celix_bundle_manifest_t* manifest = NULL;
     if (isFrameworkBundle) {
-        status = celix_bundleManifest_createFrameworkManifest(&manifest);
+        status = celix_bundleManifest_createFrameworkManifest(&archive->manifest);
     } else {
-        status = celix_bundleArchive_createCacheDirectory(archive, &manifest);
+        status = celix_bundleArchive_createCacheDirectory(archive, &archive->manifest);
     }
     if (status != CELIX_SUCCESS) {
-        error = "Failed to initialize archive or create manifest.";
-        goto dir_failed;
-    }
-
-    status = celix_bundleRevision_create(fw, archive->archiveRoot, archive->location, manifest, &archive->revision);
-    if (status != CELIX_SUCCESS) {
-        error = "Could not create bundle revision.";
-        goto revision_failed;
+        fw_logCode(fw->logger, CELIX_LOG_LEVEL_ERROR, status, "Failed to create bundle archive.");
+        if (!isFrameworkBundle) {
+            celix_utils_deleteDirectory(archive->archiveRoot, NULL);
+        }
+        return status;
     }
 
     if (!isFrameworkBundle) {
         status = celix_bundleArchive_storeBundleStateProperties(archive);
         if (status != CELIX_SUCCESS) {
-            error = "Could not store properties.";
-            goto store_prop_failed;
+            fw_logCode(fw->logger, CELIX_LOG_LEVEL_ERROR, status, "Failed to store bundle state properties.");
```

Comment:
Directory left, various resource leaks. The previous structured goto can not be removed directly.